lib/puppet/functions/archive/artifactory_checksum.rb

```diff
@@ -0,0 +1,25 @@
+require 'json'
+require 'puppet_x/bodeco/util'
+
+Puppet::Functions.create_function(:'archive::artifactory_checksum') do
+  # @summary A function that returns the checksum value of an artifact stored in Artifactory
+  # @param url The URL of the artifact.
+  # @param checksum_type The checksum type.
+  #        Note the function will raise an error if you ask for sha256 but your artifactory instance doesn't have the sha256 value calculated.
+  dispatch :artifactory_checksum do
+    param 'Stdlib::HTTPUrl', :url
+    optional_param "Enum['sha1','sha256','md5']", :checksum_type
+    return_type 'String'
```

Comment:
If I remove this line, the docs generate.
Equally, I can also add a `# @return [String] The checksum` above.
I don't think this should be required though as it doesn't add much if I'm already doing the `return_type` validation thing.

Reply:
https://tickets.puppetlabs.com/browse/PDOC-229